### ISSUE TITLE
[azure-kinect-sensor-sdk] macOS build failure

### ISSUE THREAD

**Issue by adammpolak** [microsoft/vcpkg#16329]
**Host Environment**
- OS: macOS
- Compiler: revision

**To Reproduce**
`./vcpkg install azure-kinect-sensor-sdk`

**Failure logs** 
```./vcpkg install azure-kinect-sensor-sdk
Computing installation plan...
The following packages will be built and installed:
    azure-kinect-sensor-sdk[core]:x64-osx -> 1.4.0-alpha.0#6
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/adammac/.cache/vcpkg/archives/93/930f8b5bac6d9b73761ac3276168b364138931b3.zip
Starting package 1/1: azure-kinect-sensor-sdk:x64-osx
Building package azure-kinect-sensor-sdk[core]:x64-osx...
-- Using cached /Users/adammac/dev/vcpkg-macos/vcpkg/downloads/microsoft-Azure-Kinect-Sensor-SDK-v1.4.0-alpha.0.tar.gz
-- Cleaning sources at /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/src/.0-alpha.0-a0fc9b9902.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/adammac/dev/vcpkg-macos/vcpkg/downloads/microsoft-Azure-Kinect-Sensor-SDK-v1.4.0-alpha.0.tar.gz
-- Applying patch fix-builds.patch
-- Applying patch disable-c4275.patch
-- Applying patch fix-dependency-imgui.patch
-- Applying patch add-MATROSKA_VERSION-define.patch
-- Using source at /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/src/.0-alpha.0-a0fc9b9902.clean
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /Users/adammac/dev/vcpkg-macos/vcpkg/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/bin/cmake /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/src/.0-alpha.0-a0fc9b9902.clean -DK4A_BUILD_DOCS=OFF -DBUILD_TOOLS=OFF -DK4A_SOURCE_LINK=OFF -DK4A_MTE_VERSION=ON -DBUILD_EXAMPLES=OFF -DWITH_TEST=OFF -DIMGUI_EXTERNAL_PATH=/Users/adammac/dev/vcpkg-macos/vcpkg/installed/x64-osx/include/bindings -DCMAKE_MAKE_PROGRAM=/Users/adammac/dev/vcpkg-macos/vcpkg/downloads/tools/ninja-1.10.1-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/adammac/dev/vcpkg-macos/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/adammac/dev/vcpkg-macos/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/Users/adammac/dev/vcpkg-macos/vcpkg -D_VCPKG_INSTALLED_DIR=/Users/adammac/dev/vcpkg-macos/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=x86_64 -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/adammac/dev/vcpkg-macos/vcpkg/packages/azure-kinect-sensor-sdk_x64-osx/debug
    Working Directory: /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/config-x64-osx-dbg-out.log
      /Users/adammac/dev/vcpkg-macos/vcpkg/buildtrees/azure-kinect-sensor-sdk/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:328 (vcpkg_execute_required_process)
  ports/azure-kinect-sensor-sdk/portfile.cmake:25 (vcpkg_configure_cmake)
  scripts/ports.cmake:133 (include)


Error: Building package azure-kinect-sensor-sdk:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: azure-kinect-sensor-sdk:x64-osx
  Vcpkg version: 2021-01-13-unknownhash

Additionally, attach any relevant sections from the log files above.
```

```./vcpkg install azure-kinect-sensor-sdk
Computing installation plan...
The following packages will be built and installed:
    azure-kinect-sensor-sdk[core]:x64-osx -> 1.4.0-alpha.0#6
Detecting compiler hash for triplet x64-osx...
Could not locate cached archive: /Users/adammac/.cache/vcpkg/archives/93/930f8b5bac6d9b73761ac3276168b364138931b3.zip
Starting package 1/1: azure-kinect-sensor-sdk:x64-osx
Building package azure-kinect-sensor-sdk[core]:x64-osx...
-- Using cached /Users/adammac/dev/vcpkg-macos/vcpkg/downloads/microsoft-Azure-Kinect-Sensor-SDK-v1.4.0-alpha.0.tar.gz
CMake Error at CMakeLists.txt:112 (find_library):
  Could not find LibUSB using the following names: libusb-1.0
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by adammpolak** [comment]
It cannot find libusb-1.0

**Comment by JackBoosY** [comment]
Duplicated to #12995.

**Comment by adammpolak** [comment]
@JackBoosY are we positive this is a duplicate of #12995?

I see that at one point you mentioned that it cannot find libusb: https://github.com/microsoft/vcpkg/issues/12995#issuecomment-678080450

but then later it seems like the issue there is that libuvc has 2 different versions:
https://github.com/microsoft/vcpkg/issues/12995#issuecomment-720992478

That issue does not seem to try to solve finding libusb.

**Comment by JackBoosY** [comment]
@adammpolak  Can #14750 solve this issue?

**Comment by adammpolak** [comment]
@JackBoosY if #14750  also solves the issue of https://github.com/microsoft/vcpkg/issues/12995#issuecomment-678080450 cannot find USB then I think it might.

I will try to install it again once the PR is merged. Will it be as easy as re-installing vcpkg from gitclone?

**Comment by JackBoosY** [comment]
@adammpolak Yep. And you can pull #14750 changes to test it also.

**Comment by adammpolak** [comment]
Sorry for triple checking, but this means I test by cloning this repo: https://github.com/fran6co/vcpkg/tree/azure_linux ?

Or how do I checkout this update locally (I don't want to provide wrong feedback on the fix working)

**Comment by adammpolak** [comment]
@JackBoosY I:

- cloned https://github.com/fran6co/vcpkg/
- checked out azure_linux
- ran `bash bootstrap-vcpkg.sh`
- ran `./vcpkg install azure-kinect-sensor-sdk`
- Was able to get through dependencies then run into this issue: `Could not find a package configuration file provided by "libuvc" with any of the following names` ...

Full `vcpkg/buildtrees/azure-kinect-sensor-sdk/config-x64-osx-dbg-err.log` below

```Starting package 16/16: azure-kinect-sensor-sdk:x64-osx
Building package azure-kinect-sensor-sdk[core]:x64-osx...
-- Downloading https://github.com/microsoft/Azure-Kinect-Sensor-SDK/archive/v1.4.1.tar.gz -> microsoft-Azure-Kinect-Sensor-SDK-v1.4.1.tar.gz...
-- Extracting source /Users/adammac/dev/vcpkg/downloads/microsoft-Azure-Kinect-Sensor-SDK-v1.4.1.tar.gz
-- Applying patch fix-builds.patch
-- Applying patch fix-dependency-imgui.patch
-- Applying patch add-MATROSKA_VERSION-define.patch
-- Applying patch fix-linux.patch
-- Applying patch fix-uvc.patch
-- Using source at /Users/adammac/dev/vcpkg/buildtrees/azure-kinect-sensor-sdk/src/v1.4.1-4c7ddb5afb.clean
-- Configuring x64-osx-dbg
CMake Error at /Users/adammac/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:653 (_find_package):
  Could not find a package configuration file provided by "libuvc" with any
  of the following names:

    libuvcConfig.cmake
    libuvc-config.cmake

  Add the installation prefix of "libuvc" to CMAKE_PREFIX_PATH or set
  "libuvc_DIR" to a directory containing one of the above files.  If "libuvc"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:114 (find_package)
```

It seems that this PR has an issue with libuvc (so it seems like the libusb issue has been solved?)

**Comment by JackBoosY** [comment]
At least #14750 fixed the issue about libusb, I will try to support OSX after that merged.

**Comment by adammpolak** [comment]
Thank you @JackBoosY !

**Comment by FrankXie05** [comment]
@adammpolak The upstream explicitly supports Linux and Windows. 
https://github.com/microsoft/Azure-Kinect-Sensor-SDK#introduction
